### PR TITLE
Added Hamburger Menu for Mobile Responsiveness

### PR DESCRIPTION
**Pull request by tanish0912**
### Description

- **Feature:** Added a responsive hamburger menu for mobile screens.
- **Objective:** Improve navigation experience on mobile devices by providing a compact menu that is easily accessible.
- **Implementation:**
  - Created a new hamburger menu button.
  - Implemented toggle functionality to open and close the menu.
  - Styled the menu items for better visibility and usability on smaller screens.
- **Testing:** Verified functionality on various mobile screen sizes to ensure consistency and smooth user experience.

### CODE REVIEW

**Comment by ChiragAgg5k** [comment]
Screenshot 2024-10-02 at 12 15 05 AM


@tanish0912 there are some build issues. can u confirm building the project on ur machine once?

**Comment by tanish0912** [comment]
Screenshot 2024-10-02 at 12 33 38 AM
Screenshot 2024-10-02 at 12 33 50 AM

**Comment by ChiragAgg5k** [comment]
@tanish0912 the project uses pnpm... you need to build once using pnpm which then updates the lockfile for it.

File: .gitignore [hunk 1]
```diff
@@ -34,3 +34,8 @@ yarn-error.log*
 # typescript
 *.tsbuildinfo
 next-env.d.ts
+
+
+.env.local/
```
Comment (ChiragAgg5k) [inline .gitignore:39]: the `.gitignore` file already has ignored env files:

```txt
# local env files
.env*.local
```

**Pull request closed without merging**